registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/UserNotificationApi.java

```diff
@@ -0,0 +1,316 @@
+
+	/** The core audit request builder. */
+	@Autowired
+	AuditLogRequestBuilder auditLogRequestBuilder;
+
+	@Autowired
+	UserNotificationRequestValidator UserNotificationRequestValidator;
+
+	/**
+	 * Deploy verticle.
+	 */
+	public void deployVerticle() {
+		mosipEventBus = this.getEventBus(this, clusterManagerUrl, workerPoolSize);
+	}
+
+	@Override
+	public void start() {
+		router.setRoute(this.postUrl(getVertx(), null, null));
+		this.routes(router);
+		this.createServer(router.getRouter(), Integer.parseInt(port));
+	}
+
+	@Override
+	public Integer getEventBusPort() {
+		return Integer.parseInt(eventBusPort);
+	}
+
+	/**
+	 * contains all the routes in this stage
+	 *
+	 * @param router
+	 */
+	private void routes(MosipRouter router) {
+		router.post(contextPath + "/userNotification");
+		router.handler(this::processURL, this::failure);
+	}
+
+	/**
+	 * method to process the context received.
+	 *
+	 * @param ctx the ctx
+	 */
+	public void processURL(RoutingContext ctx) {
+		LogDescription description = new LogDescription();
+		InternalRegistrationStatusDto registrationStatusDto;
+		JsonObject obj = ctx.getBodyAsJson();
+		String rid = null;
+		try {
+			UserNotificationDTO UserNotificationDTO = JsonUtil.readValueWithUnknownProperties(obj.toString(),
+					UserNotificationDTO.class);
+			UserNotificationRequestDTO UserNotificationRequestDTO = UserNotificationDTO.getRequest();
+			rid = UserNotificationRequestDTO.getRid();
+			regProcLogger.debug("UserNotificationApi:processURL called for registration id {}",
+					UserNotificationDTO.getRequest().getRid());
+			UserNotificationRequestValidator.validate(UserNotificationDTO);
+			String user = getUser(ctx);
+
+			registrationStatusDto = registrationStatusService.getRegistrationStatus(rid,UserNotificationRequestDTO.getRegType(),1,null);
+			if (registrationStatusDto == null) {
+				description.setMessage(PlatformErrorMessages.RPR_UNA_RID_NOT_FOUND.getMessage());
+				description.setCode(PlatformErrorMessages.RPR_UNA_RID_NOT_FOUND.getCode());
+				updateAudit(description, UserNotificationRequestDTO.getRid(), false, user);
+				throw new UserNotificationException(PlatformErrorMessages.RPR_UNA_RID_NOT_FOUND.getCode(),
+						String.format(PlatformErrorMessages.RPR_UNA_RID_NOT_FOUND.getMessage(), rid));
+			}
+			if (!registrationStatusDto.getRegistrationType().equals(UserNotificationRequestDTO.getRegType())) {
+				description.setMessage(PlatformErrorMessages.RPR_UNA_REGTYPE_NOT_MATCHING.getMessage());
+				description.setCode(PlatformErrorMessages.RPR_UNA_REGTYPE_NOT_MATCHING.getCode());
+				updateAudit(description, UserNotificationRequestDTO.getRid(), false, user);
+				throw new UserNotificationException(PlatformErrorMessages.RPR_UNA_REGTYPE_NOT_MATCHING.getCode(),
+						String.format(PlatformErrorMessages.RPR_UNA_REGTYPE_NOT_MATCHING.getMessage(), rid));
+				}
+				
+				sendWorkflowCompletedWebSubEvent(registrationStatusDto);
```

Comment:
Sending workflow completed websub event will not cause other actions additional to resending the notification ?

Reply:
No it will not make any other action. only publish the event to websub.